Mathlib/Algebra/Category/ModuleCat/Baer.lean

```diff
@@ -0,0 +1,134 @@
+/-
+Copyright (c) 2025 Nailin Guan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nailin Guan
+-/
+module
+
+public import Mathlib.Algebra.Category.Grp.Zero
+public import Mathlib.Algebra.Category.ModuleCat.EnoughInjectives
+public import Mathlib.Algebra.Category.ModuleCat.Projective
+public import Mathlib.Algebra.Homology.DerivedCategory.Ext.EnoughInjectives
+public import Mathlib.Algebra.Homology.DerivedCategory.Ext.EnoughProjectives
+public import Mathlib.Algebra.Homology.ShortComplex.ModuleCat
+public import Mathlib.RingTheory.Ideal.Quotient.Operations
+
+/-!
+# Category Language Baer Criterion
+-/
+
+@[expose] public section
+
+universe u u' v v'
+
+variable {R : Type u} [CommRing R]
+
+open CategoryTheory Abelian
+
+private instance small_of_quotient [Small.{v} R] (I : Ideal R) : Small.{v} (R ⧸ I) :=
+  small_of_surjective Ideal.Quotient.mk_surjective
+
+section
+
+universe w
+
+variable [UnivLE.{v, w}]
+
+private instance hasExt_of_small [Small.{v} R] : CategoryTheory.HasExt.{w} (ModuleCat.{v} R) :=
+  CategoryTheory.hasExt_of_enoughProjectives.{w} (ModuleCat.{v} R)
```

Comment:
It would be reasonable to make this PR depend on #32312, and use only `HasExt.{v}`.